File: minegame-broker/src/main/java/top/nipuru/minegame/broker/game/GamePlayerList.java
package top.nipuru.minegame.broker.game;

import lombok.extern.slf4j.Slf4j;

import java.util.Collection;
import java.util.Collections;
import java.util.Map;
import java.util.UUID;
import java.util.concurrent.ConcurrentHashMap;

@Slf4j
public class GamePlayerList {

    private final Map<UUID, GamePlayer> byUniqueId = new ConcurrentHashMap<>();
    private final Map<Integer, GamePlayer> byPlayerId = new ConcurrentHashMap<>();

    public GamePlayerList() {
    }

    public void registerPlayer(GamePlayer player) {
        if (log.isDebugEnabled()) {
            log.debug("Register GamePlayer: {}", player.getName());
        }
        byUniqueId.put(player.getUniqueId(), player);
        byPlayerId.put(player.getPlayerId(), player);
    }

    public GamePlayer getPlayer(UUID uniqueId) {
        GamePlayer gamePlayer = byUniqueId.get(uniqueId);
        if (gamePlayer == null) {
            throw new NullPointerException();
        }
        return gamePlayer;
    }

    public GamePlayer getPlayer(int playerId) {
        GamePlayer gamePlayer = byPlayerId.get(playerId);
        if (gamePlayer == null) {
            throw new NullPointerException();
        }
        return gamePlayer;
    }

    public Collection<GamePlayer> getPlayers() {
        return Collections.unmodifiableCollection(byUniqueId.values());
    }

    public void removePlayer(GamePlayer player) {
        byUniqueId.remove(player.getUniqueId());
        byPlayerId.remove(player.getPlayerId());
        if (log.isDebugEnabled()) {
            log.debug("Remove GamePlayer: {}", player.getName());
        }
    }
}

File: minegame-broker/src/main/kotlin/top/nipuru/minegame/broker/game/GamePlayerList.kt
package top.nipuru.minegame.broker.game

import top.nipuru.minegame.broker.logger
import java.util.*
import java.util.concurrent.ConcurrentHashMap

class GamePlayerList {
    private val byUniqueId: MutableMap<UUID, GamePlayer> = ConcurrentHashMap()
    private val byPlayerId: MutableMap<Int, GamePlayer> = ConcurrentHashMap()

    fun registerPlayer(player: GamePlayer) {
        if (logger.isDebugEnabled) {
            logger.debug("Register GamePlayer: {}", player.name)
        }
        byUniqueId[player.uniqueId] = player
        byPlayerId[player.playerId] = player
    }

    fun getPlayer(uniqueId: UUID): GamePlayer {
        val gamePlayer = byUniqueId[uniqueId] ?: throw NullPointerException()
        return gamePlayer
    }

    fun getPlayer(playerId: Int): GamePlayer {
        val gamePlayer = byPlayerId[playerId] ?: throw NullPointerException()
        return gamePlayer
    }

    val players: Collection<GamePlayer>
        get() = Collections.unmodifiableCollection(byUniqueId.values)

    fun removePlayer(player: GamePlayer) {
        byUniqueId.remove(player.uniqueId)
        byPlayerId.remove(player.playerId)
        if (logger.isDebugEnabled) {
            logger.debug("Remove GamePlayer: {}", player.name)
        }
    }
}
